File: youxian/src/main/java/apicloud/com/youxian/mvp/ui/activity/MainActivity.java
package apicloud.com.youxian.mvp.ui.activity;


import apicloud.com.youxian.mvp.ui.base.BaseActivity;

import javax.inject.Inject;

import apicloud.com.youxian.mvp.presenter.MainPresenter;
import apicloud.com.youxian.mvp.contract.MainContract;
import apicloud.com.youxian.component.AppComponent;
import apicloud.com.youxian.component.DaggerActivityComponent;
import apicloud.com.youxian.R;


public class MainActivity extends BaseActivity implements MainContract.View {

    public @Inject
    MainPresenter mainPresenter;

    @Override
    public void setActivityComponent(AppComponent appComponent) {
        DaggerActivityComponent //如找不到该类,请编译一下项目
                .builder()
                .appComponent(appComponent)
                .build()
                .inject(this);
    }


    @Override
    public void initView() {
        super.initView();

    }


    @Override
    public void initData() {

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mainPresenter.detachView();
    }

    @Override
    public int getLayoutRes() {
        return R.layout.activity_main;
    }

    @Override
    public void complete(String msg) {

    }

    @Override
    public void showError(String msg) {

    }
}

File: youxian/src/main/java/apicloud/com/youxian/mvp/ui/activity/MainActivity.kt
package apicloud.com.youxian.mvp.ui.activity


import apicloud.com.youxian.mvp.ui.base.BaseActivity
import javax.inject.Inject
import apicloud.com.youxian.mvp.presenter.MainPresenter
import apicloud.com.youxian.mvp.contract.MainContract
import apicloud.com.youxian.component.AppComponent
import apicloud.com.youxian.component.DaggerActivityComponent
import apicloud.com.youxian.R


class MainActivity : BaseActivity(), MainContract.View {

    @Inject lateinit var mainPresenter: MainPresenter

    override fun setActivityComponent(appComponent: AppComponent) {
        DaggerActivityComponent //如找不到该类,请编译一下项目
                .builder()
                .appComponent(appComponent)
                .build()
                .inject(this)
    }


    public override fun initView() {
        super.initView()

    }

    public override fun initData() {

    }

    override fun onDestroy() {
        super.onDestroy()
        mainPresenter.detachView()
    }


    override fun getLayoutRes(): Int = R.layout.activity_main


    override fun complete(msg: String) {

    }

    override fun showError(msg: String) {

    }
}
